src/platform/api.ts

```diff
@@ -87,20 +87,29 @@ export function buildApi(
             await service.createNew(options);
         },
         getKernelService: async () => {
-            const kernelServiceFactory = serviceContainer.get<JupyterKernelServiceFactory>(JupyterKernelServiceFactory);
+            const kernelServiceFactory =
+                serviceContainer.get<IExportedKernelServiceFactory>(IExportedKernelServiceFactory);
             return kernelServiceFactory.getService();
         }
     };
 
     // In test/dev environment return the DI Container.
     if (
         isTestExecution() ||
-        process.env.VSC_JUPYTER_EXPOSE_SVC ||
-        context.extensionMode === ExtensionMode.Development
+        context.extensionMode === ExtensionMode.Development ||
+        context.extensionMode === ExtensionMode.Test
     ) {
         /* eslint-disable @typescript-eslint/no-explicit-any */
         (api as any).serviceContainer = serviceContainer;
         (api as any).serviceManager = serviceManager;
+        (api as any).getSymbol = (symbol: String) => {
+            // Needed for web tests. Eval for the extension happens in another web worker so
```

Comment:
Not following this.
I thought the tests will run in the same context as the webworker where the extension is running.
After all, that's how we're able to access static variables and the like.
Also here we're running the same value `IApplicationShell`

I think what's happening is we might be ending up with two bundles, which results in two copies of IApplicationShell.
Optionally you can use `export const IApplicationShell = Symbol.for("IApplicationShell")` that will work everywhere.

Reply:
There are two bundles. This is how I export the value to be read in the other one.

Reply:
> After all, that's how we're able to access static variables and the like.

Not sure this is going to work based on what happened with the IApplicationShell.

Reply:
What's the difference between `Symbol('IApplicationShell') and Symbol.for('IApplicationShell')`

Reply:
With `Symbol.for` the symbol is unique across the app domain.

Reply:
>  This is how I export the value to be read in the other one.

That explains the reason for the issue.

Reply:
> This is how I export the value to be read in the other one.

Why do we have two bundles? Can't the test code reference the JS files from the worker?
I would have expected they could, after all its all running in the same execution context.

Reply:
Look here for more information:
https://github.com/microsoft/vscode-test-web

That's how the test code works. It loads the test bundle in a server and loads the extension in webworker.